src/Init/Data/BitVec/Bitblast.lean

```diff
@@ -1280,4 +1280,62 @@ theorem getMsbD_umod {n d : BitVec w}:
     simp [BitVec.getMsbD_eq_getLsbD, hi]
   · simp [show w ≤ i by omega]
 
+/- ### smod -/
+
+theorem getElem_smod {n d : BitVec w} (hi : i < w) :
+    (BitVec.smod n d)[i]
+      = (match n.msb, d.msb with
+    | false, false => (n.umod d)
+    | false, true =>
+      let u := n.umod (-d);
+      if u = 0#w then u else (u + d)
+    | true, false =>
+      let u := (-n).umod d;
+      if u = 0#w then u else (d - u)
+    | true, true => (-(-n).umod (-d)))[i] := by
+  rw [smod_eq]
+  cases n.msb  <;> cases d.msb <;> simp
```

Comment:
The RHS of this theorem (and all of the ones below) are not particularly pretty.

For the sake of the completeness of the API, I'd be happy to create a new `def smodCircuit := ...`, which hides the reduction from `smod` to `umod`, and reshape the right hand sides of all of these theorems into this `def`.